packages/gatsby/src/db/loki/nodes-query.js

```diff
@@ -244,28 +294,71 @@ function ensureFieldIndexes(coll, lokiArgs) {
  * @returns {promise} A promise that will eventually be resolved with
  * a collection of matching objects (even if `firstOnly` is true)
  */
-async function runQuery({ gqlType, queryArgs, firstOnly }) {
+async function runQuery(args) {
+  if (args.nodeTypeNames.length > 1) {
+    const nodes = args.nodeTypeNames.reduce(
+      (acc, typeName) => acc.concat(getNodesByType(typeName)),
+      []
+    )
+    return runSiftOnNodes(nodes, args, getNode)
```

Comment:
Do we have to fall back on sift here? can't we query each node type collection separately and then concatenate and sort the results? this way we could potentially benefit from loki indices when filtering, and had a better separation between the backends.

Reply:
That is a good idea, but let's add it in future PR.